server/src/internalClusterTest/java/org/opensearch/indices/stats/IndexStatsIT.java

```diff
@@ -652,122 +660,124 @@ public void testSimpleStats() throws Exception {
+            assertThat(stats.getIndex("test1").getTotal().getDocs().getCount(), equalTo(test1ExpectedWrites));
+            assertThat(stats.getIndex("test1").getPrimaries().getStore(), notNullValue());
+            assertThat(stats.getIndex("test1").getPrimaries().getMerge(), notNullValue());
+            assertThat(stats.getIndex("test1").getPrimaries().getFlush(), notNullValue());
+            assertThat(stats.getIndex("test1").getPrimaries().getRefresh(), notNullValue());
+
+            assertThat(stats.getIndex("test2").getPrimaries().getDocs().getCount(), equalTo(1L));
+            assertThat(stats.getIndex("test2").getTotal().getDocs().getCount(), equalTo(test2ExpectedWrites));
+
+            // make sure that number of requests in progress is 0
+            assertThat(stats.getIndex("test1").getTotal().getIndexing().getTotal().getIndexCurrent(), equalTo(0L));
+            assertThat(stats.getIndex("test1").getTotal().getIndexing().getTotal().getDeleteCurrent(), equalTo(0L));
+            assertThat(stats.getIndex("test1").getTotal().getSearch().getTotal().getFetchCurrent(), equalTo(0L));
+            assertThat(stats.getIndex("test1").getTotal().getSearch().getTotal().getQueryCurrent(), equalTo(0L));
+            assertThat(stats.getIndex("test2").getTotal().getIndexing().getTotal().getIndexCurrent(), equalTo(0L));
+            assertThat(stats.getIndex("test2").getTotal().getIndexing().getTotal().getDeleteCurrent(), equalTo(0L));
+            assertThat(stats.getIndex("test2").getTotal().getSearch().getTotal().getFetchCurrent(), equalTo(0L));
+            assertThat(stats.getIndex("test2").getTotal().getSearch().getTotal().getQueryCurrent(), equalTo(0L));
+
+            // check flags
+            stats = client().admin().indices().prepareStats().clear().setFlush(true).setRefresh(true).setMerge(true).execute().actionGet();
+
+            assertThat(stats.getTotal().getDocs(), nullValue());
+            assertThat(stats.getTotal().getStore(), nullValue());
+            assertThat(stats.getTotal().getIndexing(), nullValue());
+            assertThat(stats.getTotal().getMerge(), notNullValue());
+            assertThat(stats.getTotal().getFlush(), notNullValue());
+            assertThat(stats.getTotal().getRefresh(), notNullValue());
+
+            // check get
+            GetResponse getResponse = client().prepareGet("test2", "1").execute().actionGet();
+            assertThat(getResponse.isExists(), equalTo(true));
 
-        stats = client().admin().indices().prepareStats().execute().actionGet();
-        assertThat(stats.getTotal().getGet().getCount(), equalTo(2L));
-        assertThat(stats.getTotal().getGet().getExistsCount(), equalTo(1L));
-        assertThat(stats.getTotal().getGet().getMissingCount(), equalTo(1L));
+            stats = client().admin().indices().prepareStats().execute().actionGet();
+            assertThat(stats.getTotal().getGet().getCount(), equalTo(1L));
```

Comment:
why are we changing values of the count?

Reply:
looks like they got in by mistake. I will update this

Reply:
Looks like I haven't changed these, the git diff shows like values have been changed, but actually they haven't. It was just refactoring same code blocks up and down. The git diff makes the changes a little confusing, but no values of count are changed here.